truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayGuards.java

```diff
@@ -73,4 +73,30 @@ public static boolean areBothObject(RubyArray a, RubyArray b) {
         return a.getStore() instanceof Object[] && b.getStore() instanceof Object[];
     }
 
+    // New names
+
+    public static boolean isNullArray(RubyArray array) {
+        return array.getStore() == null;
+    }
+
+    public static boolean isIntegerArray(RubyArray array) {
```

Comment:
What about isIntArray? So mapping from primitive type to guard name is just "is#{type.capitalize}Array".